### PR TITLE
Document `cockroach workload` and expand `cockroach demo` docs

### PR DESCRIPTION
**Pull request by jseldess**
- Document `cockroach workload` command.
- Load initial datasets with `cockroach demo` 
- Use built-in workloads in tutorials.

@danhhz, please review the first two commits. @petermattis or @bdarnell, please review the third commit.

Once I get approval, I'll apply these changes to the v2.2 docs as well.

### CODE REVIEW

**Comment by jseldess** [comment]
Fixes https://github.com/cockroachdb/cockroach/issues/32398.

File: v2.1/demo-automatic-rebalancing.md [hunk 1]
```diff
@@ -122,42 +117,48 @@ $ cockroach sql --execute="SHOW ZONE CONFIGURATION FOR RANGE default;" --insecur
...
+          8s        0          337.5          364.6     21.0     50.3     75.5    100.7 transfer
+          9s        0          349.4          362.9     21.0     46.1     62.9    117.4 transfer
+         10s        0          323.1          358.9     23.1     48.2     67.1    100.7 transfer
+    ...
+    ~~~
 
 ## Step 6. Watch the replica count increase
 
-Open the Admin UI at `http://localhost:8080` and you’ll see the bytes, replica count, and other metrics increase as the `block_writer` program inserts data.
+Open the Admin UI at `http://localhost:8080` and you’ll see the bytes, replica count, and other metrics increase as the `tpcc` workload writes data.
```
Comment (bdarnell) [inline v2.1/demo-automatic-rebalancing.md]: This tutorial is running `bank`, not `tpcc`. 

`bank` doesn't generate much data, so the replica count will grow slowly, if at all. It's not a great fit for this tutorial. Maybe it should use tpcc (with 10 or 100 warehouses) instead, although the setup step might complicate things. None of the other built-in workloads generate a lot of data.

Comment (danhhz) [reply v2.1/demo-automatic-rebalancing.md]: Should we include kv in the workloads shipped in the cockroach binary?

Comment (bdarnell) [reply v2.1/demo-automatic-rebalancing.md]: Yes, I think so.

**Comment by jseldess** [comment]
___
*[v2.1/demo-automatic-rebalancing.md, line 161 at r1](https://reviewable.io/reviews/cockroachdb/docs/4087#-LRnLzib2WErn-Gz9icc-r1-161:-LSR_38v8k3gcXjGbQgY:bgv0809) ([raw file](https://github.com/cockroachdb/docs/blob/d83cc60905dbc616e1c03e60c183756bceec03e5/v2.1/demo-automatic-rebalancing.md#L161)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Yes, I think so. 
</blockquote></details>

Done. PTAL, @bdarnell. Note that I had to add a callout due to https://github.com/cockroachdb/cockroach/issues/32699.

**Comment by jseldess** [comment]
___
*[v2.1/cockroach-workload.md, line 75 at r2](https://reviewable.io/reviews/cockroachdb/docs/4087#-LSRZEXnA4A5ylGL_oSU:-LSRc-Ag6NmdRpNO706X:bmazzhy) ([raw file](https://github.com/cockroachdb/docs/blob/bb2f9d5ddac0be8d2dc28f4863bf4cb64a9c2064/v2.1/cockroach-workload.md#L75)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

nit: missing backtick before --rows
</blockquote></details>

Thanks for catching. Done.

**Comment by jseldess** [comment]
@bdarnell gave me the go-ahead in another channel, so merging this.

**Pull request merged**